packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -303,6 +399,23 @@ export class Table extends Construct {
             });
         }
     }
+
+    private registerNonKeyAttributes(nonKeyAttributes: string[]) {
+        if (this.attributeDefinitions.length + this.nonKeyAttributes.length + nonKeyAttributes.length > 20) {
+            // https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-secondary-indexes
+            throw new RangeError('a maximum number of nonKeyAttributes across all of secondary indexes is 20');
+        }
+
+        // store all non-key attributes
+        nonKeyAttributes.forEach(nonKeyAttribute => this.nonKeyAttributes.push(nonKeyAttribute));
+
+        // throw error if key attribute is part of non-key attributes
+        this.attributeDefinitions.forEach(keyAttribute => {
+            if (typeof keyAttribute.attributeName === 'string' && this.nonKeyAttributes.includes(keyAttribute.attributeName)) {
+                throw new Error(`${keyAttribute.attributeName} should not be part of ${this.nonKeyAttributes}`);
```

Comment:
Explain why in the error message

Reply:
All key attributes are added by default when using `Include` projection type, and CloudFormation fails to create a stack if a key attribute is part of non-key attributes.

I will reword the error message in a more descriptive way.